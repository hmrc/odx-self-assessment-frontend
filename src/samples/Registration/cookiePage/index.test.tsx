import React from 'react';
import { cleanup, render } from '@testing-library/react';
import { I18nextProvider, useTranslation } from 'react-i18next';
import { renderHook } from '@testing-library/react-hooks';
import { act } from 'react-dom/test-utils';
import CookiePage from '.';
import { mockGetSdkConfigWithBasepath } from '../../../../tests/mocks/getSdkConfigMock';

jest.mock('@pega/auth/lib/sdk-auth-manager', () => ({
  getSdkConfig: jest.fn()
}));

describe('should render CookiePage component.', () => {
  let t;
  afterEach(cleanup);

  beforeEach(async () => {
    mockGetSdkConfigWithBasepath();
    t = renderHook(() => useTranslation());

    await act(async () => {
      t.result.current.i18n.changeLanguage('en');
    });
  });

  it('should render CookiePage content in English.', async () => {
    let asFragment;

    await act(async () => {
      const component = render(
        <I18nextProvider i18n={t.result.current.i18n}>
          <CookiePage />
        </I18nextProvider>
      );
      asFragment = component.asFragment;
    });

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render CookiePage content in Welsh.', async () => {
    let asFragment;

    await act(async () => {
      t.result.current.i18n.changeLanguage('cy');

      const component = render(
        <I18nextProvider i18n={t.result.current.i18n}>
          <CookiePage />
        </I18nextProvider>
      );
      asFragment = component.asFragment;
    });

    expect(asFragment()).toMatchSnapshot();
  });
});
