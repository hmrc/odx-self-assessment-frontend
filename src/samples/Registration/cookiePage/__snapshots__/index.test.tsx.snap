// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CookiePage component. should render CookiePage content in English. 1`] = `
<DocumentFragment>
  <h1
    class="govuk-heading-l"
  >
    Cookies
  </h1>
  <p
    class="govuk-body"
  >
    Cookies are small files saved on your phone, tablet or computer when you visit a website.
  </p>
  <p
    class="govuk-body"
  >
    We use the following cookies to make this site works and collect information about how you use our service.
  </p>
  <h2
    class="govuk-heading-m"
  >
    Essential Cookies
  </h2>
  <p
    class="govuk-body"
  >
    Essential cookies keep your information secure whilst you use this service. We do not need to ask permission to use them.
  </p>
  <table
    class="govuk-table"
  >
    <thead
      class="govuk-table__head"
    >
      <tr
        class="govuk-table__row"
      >
        <th
          class="govuk-table__header"
          scope="col"
        >
          Name
        </th>
        <th
          class="govuk-table__header"
          scope="col"
        >
          Purpose
        </th>
        <th
          class="govuk-table__header"
          scope="col"
        >
          Expires
        </th>
      </tr>
    </thead>
    <tbody
      class="govuk-table__body"
    >
      <tr
        class="govuk-table__row"
      >
        <td
          class="govuk-table__cell"
        >
          AWSALB
        </td>
        <td
          class="govuk-table__cell"
        >
          Used to help manage web traffic between a visitors web browser and the application servers that host HMRC services.
        </td>
        <td
          class="govuk-table__cell"
        >
          7 days
        </td>
      </tr>
      <tr
        class="govuk-table__row"
      >
        <td
          class="govuk-table__cell"
        >
          AWSALBCORS
        </td>
        <td
          class="govuk-table__cell"
        >
          Used to help manage web traffic between a visitors web browser and the application servers that host HMRC services.
        </td>
        <td
          class="govuk-table__cell"
        >
          7 days
        </td>
      </tr>
      <tr
        class="govuk-table__row"
      >
        <td
          class="govuk-table__cell"
        >
          JSESSIONID
        </td>
        <td
          class="govuk-table__cell"
        >
          Used to identify the visitor’s session information related to the HMRC service being accessed.
        </td>
        <td
          class="govuk-table__cell"
        >
          Session
        </td>
      </tr>
      <tr
        class="govuk-table__row"
      >
        <td
          class="govuk-table__cell"
        >
          PEGAODXEI
        </td>
        <td
          class="govuk-table__cell"
        >
          This is used to identify the session ID of a user not using Government Gateway.
        </td>
        <td
          class="govuk-table__cell"
        >
          10 years
        </td>
      </tr>
      <tr
        class="govuk-table__row"
      >
        <td
          class="govuk-table__cell"
        >
          PEGAODXDI
        </td>
        <td
          class="govuk-table__cell"
        >
          This is used to uniquely identify a device that has interacted with HMRC services and is used for internal security auditing.
        </td>
        <td
          class="govuk-table__cell"
        >
          10 years
        </td>
      </tr>
      <tr
        class="govuk-table__row"
      >
        <td
          class="govuk-table__cell"
        >
          AKA_A2
        </td>
        <td
          class="govuk-table__cell"
        >
          This Akamai cookie is used to help speed up content delivery on websites using adaptive acceleration.
        </td>
        <td
          class="govuk-table__cell"
        >
          1 hour
        </td>
      </tr>
    </tbody>
  </table>
  <p
    class="govuk-body"
  >
    <a
      class="govuk-link govuk-link--no-visited-state"
      href="https://www.tax.service.gov.uk/help/cookie-details"
    >
      Find out more about cookies on HMRC services
    </a>
  </p>
</DocumentFragment>
`;

exports[`should render CookiePage component. should render CookiePage content in Welsh. 1`] = `
<DocumentFragment>
  <h1
    class="govuk-heading-l"
  >
    Cwcis
  </h1>
  <p
    class="govuk-body"
  >
    Cwcis yw ffeiliau bach sy’n cael eu cadw ar eich ffôn, llechen neu gyfrifiadur pan ydych yn ymweld â gwefan.
  </p>
  <p
    class="govuk-body"
  >
    Rydym yn defnyddio’r cwcis canlynol i wneud yn siŵr bod y wefan hon yn gweithio ac yn casglu gwybodaeth am sut rydych chi’n defnyddio ein gwasanaeth
  </p>
  <h2
    class="govuk-heading-m"
  >
    Cwcis Hanfodol
  </h2>
  <p
    class="govuk-body"
  >
    Mae cwcis hanfodol yn cadw eich gwybodaeth yn ddiogel tra byddwch yn defnyddio’r gwasanaeth hwn. Nid oes angen i ni ofyn am ganiatâd i’w defnyddio
  </p>
  <table
    class="govuk-table"
  >
    <thead
      class="govuk-table__head"
    >
      <tr
        class="govuk-table__row"
      >
        <th
          class="govuk-table__header"
          scope="col"
        >
          Enw
        </th>
        <th
          class="govuk-table__header"
          scope="col"
        >
          Pwrpass
        </th>
        <th
          class="govuk-table__header"
          scope="col"
        >
          Yn dod i ben
        </th>
      </tr>
    </thead>
    <tbody
      class="govuk-table__body"
    >
      <tr
        class="govuk-table__row"
      >
        <td
          class="govuk-table__cell"
        >
          AWSALB
        </td>
        <td
          class="govuk-table__cell"
        >
          Used to help manage web traffic between a visitors web browser and the application servers that host HMRC services.
        </td>
        <td
          class="govuk-table__cell"
        >
          7 days
        </td>
      </tr>
      <tr
        class="govuk-table__row"
      >
        <td
          class="govuk-table__cell"
        >
          AWSALBCORS
        </td>
        <td
          class="govuk-table__cell"
        >
          Used to help manage web traffic between a visitors web browser and the application servers that host HMRC services.
        </td>
        <td
          class="govuk-table__cell"
        >
          7 days
        </td>
      </tr>
      <tr
        class="govuk-table__row"
      >
        <td
          class="govuk-table__cell"
        >
          JSESSIONID
        </td>
        <td
          class="govuk-table__cell"
        >
          Used to identify the visitor’s session information related to the HMRC service being accessed.
        </td>
        <td
          class="govuk-table__cell"
        >
          Session
        </td>
      </tr>
      <tr
        class="govuk-table__row"
      >
        <td
          class="govuk-table__cell"
        >
          PEGAODXEI
        </td>
        <td
          class="govuk-table__cell"
        >
          Defnyddir hyn er mwyn adnabod yr ID sesiwn ar gyfer defnyddiwr nad yw’n defnyddio Porth y Llywodraeth.
        </td>
        <td
          class="govuk-table__cell"
        >
          10 mlynedd
        </td>
      </tr>
      <tr
        class="govuk-table__row"
      >
        <td
          class="govuk-table__cell"
        >
          PEGAODXDI
        </td>
        <td
          class="govuk-table__cell"
        >
          Defnyddir hyn er mwyn adnabod dyfais benodol sydd wedi rhyngweithio â gwasanaethau CThEF. Fe’i defnyddir at ddiben archwiliadau diogelwch mewnol.
        </td>
        <td
          class="govuk-table__cell"
        >
          10 mlynedd
        </td>
      </tr>
      <tr
        class="govuk-table__row"
      >
        <td
          class="govuk-table__cell"
        >
          AKA_A2
        </td>
        <td
          class="govuk-table__cell"
        >
          Defnyddir y cwci Akamai hwn i helpu cyflymu’r ddarpariaeth o gynnwys ar wefannau sy’n defnyddio cyflymiad addasol.
        </td>
        <td
          class="govuk-table__cell"
        >
          1 awr
        </td>
      </tr>
    </tbody>
  </table>
  <p
    class="govuk-body"
  >
    <a
      class="govuk-link govuk-link--no-visited-state"
      href="https://www.tax.service.gov.uk/help/cookie-details"
    >
      Find out more about cookies on HMRC services
    </a>
  </p>
</DocumentFragment>
`;
